pkg/agent/inteface.go

```diff
@@ -31,7 +31,8 @@ type AgentAddon interface {
 	//   - the RBAC permission bond to the addon agents *in the managed cluster*. (the hub cluster's RBAC
 	//     setup shall be done at GetAgentAddonOptions below.)
 	// NB for dispatching namespaced resources, it's recommended to include the namespace in the list.
-	Manifests(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn) ([]runtime.Object, error)
+	// Note that hostingCluster should be set to nil when not deploying in hosted mode.
+	Manifests(cluster *clusterv1.ManagedCluster, addon *addonapiv1alpha1.ManagedClusterAddOn, hostingCluster *clusterv1.ManagedCluster) ([]runtime.Object, error)
```

Comment:
this is a breaking change. I would rather not directly add a hostingCluster var since it is not required for all addons. Probably you could have a ops var here and allow different options.